### ISSUE TITLE
multiple GPU trainning sft error

### ISSUE THREAD

**Issue by HaughtonGuilheremeZhang** [modelscope/ms-swift#4219]
using two A6000 in trainning sft. It raised error "../aten/src/ATen/native/cuda/Loss.cu:250: nll_loss_forward_reduce_cuda_kernel_2d: block: [0,0,0], thread: [31,0,0] Assertion t >= 0 && t < n_classes failed." The nvidia driver version is 535.230.02. cuda version is 12.4. One GPU could work but when resume the one gpu sft result in grpo trainning (async mode) the trainning reward is very low compare to the other devices using the same data and same reward policy.

**Comment by Jintao-Huang** [comment]
Please share the training script.

**Comment by HaughtonGuilheremeZhang** [comment]
> Please share the training script.
训练脚本如下：

CUDA_VISIBLE_DEVICES=0,1 \
swift sft \
  --model /mnt/sda/app/work/models/Qwen3-8B \
  --train_type lora \
  --dataset ./train_SFT_202505141545.jsonl \
  --val_dataset ./train_SFT_202505141545.jsonl \
  --gradient_checkpointing_kwargs '{"use_reentrant": false}' \
  --torch_dtype bfloat16 \
  --num_train_epochs 4 \
  --per_device_train_batch_size 4 \
  --per_device_eval_batch_size 4 \
  --learning_rate 3e-4 \
  --lora_rank 32 \
  --lora_alpha 64 \
  --lora_dropout 0.05 \
  --target_modules all-linear \
  --gradient_accumulation_steps 1 \
  --eval_steps 30 \
  --save_steps 30 \
  --save_total_limit 6 \
  --logging_steps 5 \
  --max_length 4096 \
  --output_dir /mnt/sda/app/haughton/output/sft \
  --system 'You are a helpful assistant.' \
  --warmup_ratio 0.05 \
  --dataloader_num_workers 1 \
  --save_safetensors true \
  --model_author swift \
  --model_name swift-robot \
  --neftune_noise_alpha 5 \
  --report_to wandb > "./train_sft.log" 2>&1 &

**Comment by HaughtonGuilheremeZhang** [comment]
It's the problem of GPU drivers. When the driver is updated to 570.144., It can run normally.

**Comment by sunxiaoyu12** [comment]
I have the same problem:
